include/revng/ABI/Definition.h

```diff
@@ -141,6 +141,11 @@ type: struct
       call. If equal to `true`, it's the callee, otherwise it the caller.
     type: bool
 
+  - name: FloatsAlwaysUseGenericRegisters
+    doc: |
+      Setting this to true disables usage of vector registers entirely.
+    type: bool
```

Comment:
This sounds like a property of the architecture, not of the ABI.
We should also check if this is true for other parts of `Definition.h`.

Reply:
It is a property of the ABI. MIPS spec clearly describes the vector register usage. Except GCC doesn't care (I guess their MIPS ABI implementation is about as good as their Microsoft ones is), so this is pretty much a workaround to tell "hey, ignore the vector data". I think it's viable to add this as an ABI configuration option though. We might also want to reuse for stuff like "soft-float" ARM or some vector-less i386 ABIs. You get the point.

Reply:
1. When do we use "vector" and when "float"? Are we being coherent?
2. Can we call this `PassFloatThroughGPRs`? "Generic registers" is not a term we use AFAIK.

Reply:
1. `vector` is only ever applied to registers. It just happens that on all the major architectures "float"s re-use "vector" infrastructure. In this case we're talking about _float_ values using _generic_ registers, as opposed to _vector_ ones, which is why I explicitly mentioned _float_ which I only do when forced to.

> `PassFloatThroughGPRs`

2. My only complaint the `pass` term. Since it also applies to return values.

`AlwaysUseGPRsForFloats`?

Reply:
Works!

Reply:
On the second thought, just `UseGPRsForFloats` sounds better (and is more in line with your original suggestion), let's go with that instead.

Ah, ok, but `imperative` clashes with all the adjacent "bool"s. Let's go with `FloatsUseGPRs` after all - for consistency with the other variables (If you want a wider naming overhaul, let's do it in a separate PR, this one is already way too big).

Reply:
Works.

Reply:
Done.